### DISCUSSION TITLE
Version 1.3.0-rcN release testing/observations/notes.

### DISCUSSION THREAD

**Discussion by tparchambault** [ctc-oss/fapolicy-analyzer#996] [Developers]
Starting the V1.3.0-rc1 testing with the rpms available under GitHub releases page.

**Comment by tparchambault** [comment]
Testing V1.3.0-rc1-1 over FC39, focusing on the following runtime updates:

* Add validate config button by @jw3 in https://github.com/ctc-oss/fapolicy-analyzer/pull/979
* Reload rules while profiling by @jw3 in https://github.com/ctc-oss/fapolicy-analyzer/pull/990
* Add lang for config by @jw3 in https://github.com/ctc-oss/fapolicy-analyzer/pull/991

Many of this release's other changes are related to CI, builds, and dependencies. General use should validate most of these changes.

**Comment by tparchambault** [comment]
During V1.3 testing, I noticed that some incorrect string values were passing the config validation test. Verified w/ @jw3 and in the src that only the first round of syslog and watch_fs validation has been realized: Pls see https://github.com/ctc-oss/fapolicy-analyzer/issues/997#top

**Comment by tparchambault** [comment]
Continuing w/FC39:
1. General testing
  * Installation went fine. No observed issues.
  * FC39 update occurred post install, resulting in the known inconsistencies between the rpm and fapd Trust databases. Could we alert the user about the need to have fapd rebuild its dbase (via timestamps?) if using the fapolicy-analyzer with an inactive daemon? Yep, another enhancement request.
2. Config Editor testing. 
  * `Config Differences` pop-up is a nice feature. Suggest high-lighting specific deltas, if that's not already on the enhancement list.
3. Profiler testing
  * Copied the trusted `/usr/bin/ls` to the untrusted `/tmp/my-ls.` The post profiling analysis view doesn't provide an obvious (to me) indication that its execution would be denied with the current rule default rule set, i.e. `/tmp/my-ls` has the green background high-lighting, although it is untrusted (or should be.)

**Reply by jw3** [reply]
> Copied the trusted /usr/bin/ls to the untrusted /tmp/my-ls. The post profiling analysis view doesn't provide an obvious (to me) indication that its execution would be denied with the current rule default rule set, i.e. /tmp/my-ls has the green background high-lighting, although it is untrusted (or should be.)

Was it denied?  Post the fapolicyd log output.

**Comment by tparchambault** [comment]
It was denied when running fapolicyd –debug-deny from the CL. It ran within the profiler, but that’s OK isn’t it, because the profiler run is ‘permissive’ iirc. Maybe we want ‘permissive’ to be optional? In any event, the analysis view showed my the files that /tmp/my-ls accessed.

I’ll do some image captures tomorrow AM.

I was trying to test the dynamic rules reload but didn’t get to the point of modifying them yet. I thought something should make obvious that /tmp/my-ls is untrusted. And then I saw the current rules should deny its execution; That’s when I ran fapd from the CL w/--debug-deny which did deny /tmp/my-ls’s execution.

**Reply by jw3** [reply]
> I thought something should make obvious that /tmp/my-ls is untrusted.

The analyzer view should behave as it always has, where if denied it should mark red and indicate the trust status.

See the user guide

https://github.com/ctc-oss/fapolicy-analyzer/wiki/User-Guide#policy-analyzer

If it appears broken please share the logs from the profiler run.

**Reply by tparchambault** [reply]
Logs look good. It's a UX thing IMHO. We don't clearly emphasize subject execution denials in the fapa. I'll post images below.
[UntrustedDenyProfilerLogs.zip](https://github.com/ctc-oss/fapolicy-analyzer/files/14142441/UntrustedDenyProfilerLogs.zip)

**Comment by tparchambault** [comment]
Two CL sessions on an FC39 TestBed. The smaller pane copied `/usr/bin/ls` to `/tmp/my-ls`; Executed `/tmp/my-ls` from the cmdline w/o `fapolicyd` executing, w/`fapolicyd` executing via systemctl, and lastly w/`fapolicyd` executing directly from the cmdline with the `--debug-deny` option. The larger pane was this last `my-ls` execution scenario using `$ fapolicyd --debug-deny` showing the daemon's CL output.

![fapd_cl_debug_deny_session](https://github.com/ctc-oss/fapolicy-analyzer/assets/36828847/1217e329-95d4-4426-87a9-1ae3b74da84d)

**Reply by jw3** [reply]
I'm interested in seeing
1. The profiler configuration
2. The profiler log output at the `my-ls` decision
3. The analyzer view of the logs

I want to see (3) related to your above comment
   - "I thought something should make obvious that /tmp/my-ls is untrusted."


edit:  I see what we are comparing, the above is what I want to see. Post the images and we can chat on the phone if needed.

**Reply by tparchambault** [reply]
The equivalent operation from within the `fapolicy-analyzer` doesn't emphasize that an execution denial has/or  would have occurred. This is not a bug wrt V1.3 but as I wanted to test certain rules wrt Trusted executables, and was trying to determine how that reloaded rule would be presented within the fapa.

The profiling tool with input and scrolled results:

![fapa_untrusted_profiling_1of2](https://github.com/ctc-oss/fapolicy-analyzer/assets/36828847/ff45a6fd-6044-439b-8c61-5269b8dadef7)

![fapa_untrusted_profiling_2of2](https://github.com/ctc-oss/fapolicy-analyzer/assets/36828847/6ab36090-c4a2-49dd-9a83-3be734ef66a0)

Navigating to the analysis of that execution run. Note the green high-lighting with allowed access of the `/tmp/my-ls` Subject:

![fapa_untrusted_green](https://github.com/ctc-oss/fapolicy-analyzer/assets/36828847/ee0b4a94-2153-4c46-81aa-f0297e67dbe9)

Ok then... How about from the perspective of Python? While `/tmp/my-ls` is noted as untrusted, access is still granted according to the Object listing, which is probably OK.

![fapa_untrusted_nonobvious_deny](https://github.com/ctc-oss/fapolicy-analyzer/assets/36828847/6aefe90d-d4fa-422e-8fe9-e3e9ce79897c)

So, this is a lot of words and pictures; maybe this can be addressed by addressing the Subjects background high-lighting when they are the subject of a denial?

I'll play around with the analysis view some more. Maybe flipping the columns will give me the `obvious` UX I'm looking for.

**Comment by jw3** [comment]
The log says rule 14 allowed the execution, and the trust is showing that it is Untrusted

**Comment by tparchambault** [comment]
Another image of the above analysis view with no object selected:

![fapa_untrusted_nonobvious_deny_2](https://github.com/ctc-oss/fapolicy-analyzer/assets/36828847/4d781f84-5ede-4754-8e9e-645c37056c59)

**Reply by jw3** [reply]
Thanks, Ill add an issue for this and link it back

**Comment by tparchambault** [comment]
@jw3 Thx for the help.

**Comment by tparchambault** [comment]
Over FC39, when attempting to modify and deploy the rules to deny access to a back up copy of `.bashrc` , the deployment failed. Tailing the audit.log showed that SELinux was denying the removal and/or write of `fapolicyd.pid` 

```
type=AVC msg=audit(1706912901.383:345): avc:  denied  { unlink } for  pid=3707 comm="fapolicyd" name="fapolicyd.pid" dev="tmpfs" ino=1736 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=unconfined_u:object_r:var_run_t:s0 tclass=file permissive=0
type=AVC msg=audit(1706912901.386:346): avc:  denied  { write } for  pid=3709 comm="fapolicyd" name="fapolicyd.pid" dev="tmpfs" ino=1736 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=unconfined_u:object_r:var_run_t:s0 tclass=file permissive=0
type=AVC msg=audit(1706912948.068:347): avc:  denied  { unlink } for  pid=3709 comm="fapolicyd" name="fapolicyd.pid" dev="tmpfs" ino=1736 scontext=system_u:system_r:fapolicyd_t:s0 tcontext=unconfined_u:object_r:var_run_t:s0 tclass=file permissive=0
```
This doesn't currently make too much sense to me, given that earlier profiler runs today ran fine. Maybe the deployment execution environment's pid file location warrants the SELinux denial, but not the Profiler's pid file location. Or user error, with toggling back and forth between Profiler and Rule editor views. Will investigate further, but there is the implication that we may need to ship and/or update an SELinux policy config file. I don't know how we currently address SELinux policy from the `fapolicy-analyzer` perspective.

**Reply by jw3** [reply]
> I don't know how we currently address SELinux policy from the fapolicy-analyzer perspective.

Yes this needs defined and documented, added #1000

**Comment by jw3** [comment]
I dont see any issues at this point that cant be addressed next development cycle.  Going to tag a release.

**Comment by tparchambault** [comment]
Testing over RHEL9. Had an issue w/installation of the el9 rpm:

```
[toma@localhost Development]$ sudo dnf install ./fapolicy-analyzer-1.3.0.rc1-1.el9.x86_64.rpm 
[sudo] password for toma: 
Updating Subscription Management repositories.
Last metadata expiration check: 2:26:41 ago on Wed 07 Feb 2024 11:10:09 AM EST.
Error: 
 Problem: conflicting requests
  - nothing provides gnome-icon-theme needed by fapolicy-analyzer-1.3.0~rc1-1.el9.x86_64
  - nothing provides gtksourceview3 needed by fapolicy-analyzer-1.3.0~rc1-1.el9.x86_64
  - nothing provides python3-configargparse needed by fapolicy-analyzer-1.3.0~rc1-1.el9.x86_64
  - nothing provides python3-events needed by fapolicy-analyzer-1.3.0~rc1-1.el9.x86_64
  - nothing provides python3-importlib-metadata needed by fapolicy-analyzer-1.3.0~rc1-1.el9.x86_64
  - nothing provides python3-more-itertools needed by fapolicy-analyzer-1.3.0~rc1-1.el9.x86_64
  - nothing provides python3-rx needed by fapolicy-analyzer-1.3.0~rc1-1.el9.x86_64
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
[toma@localhost Development]$
```

Attempted with both  `--nobest` and `--skip-broken` w/o success.

**Reply by tparchambault** [reply]
Come to think of it, probably should add `epel` repo... That was the issue.

On RHEL9, to add the EPEL repo:
```
$ sudo subscription-manager repos --enable codeready-builder-for-rhel-9-x86_64-rpms
$ sudo dnf install https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
```

**Comment by tparchambault** [comment]
On RHEL9, observed no new issues.
